gallery/plot_scripted_tensor_transforms.py

```diff
@@ -0,0 +1,144 @@
+transformations on Tensor images. In particular, we show how image transforms
+can be performed on GPU, and how one can also script them using JIT compilation.
+
+Prior to v0.8.0, transforms in torchvision have traditionally been PIL-centric
+and presented multiple limitations due to that. Now, since v0.8.0, transforms
+implementations are Tensor and PIL compatible and we can achieve the following
+new features:
+
+- transform multi-band torch tensor images (with more than 3-4 channels)
+- torchscript transforms together with your model for deployment
+- support for GPU acceleration
+- batched transformation such as for videos
+- read and decode data directly as torch tensor with torchscript support (for PNG and JPEG image formats)
+
+.. note::
+    These features are only possible with **Tensor** images.
+"""
+
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import torch
+import torchvision.transforms as T
+from torchvision.io import read_image
+
+
+plt.rcParams["savefig.bbox"] = 'tight'
+torch.manual_seed(0)
+
+
+def show(imgs):
+    fix, axs = plt.subplots(ncols=len(imgs), squeeze=False)
+    for i, img in enumerate(imgs):
+        img = T.ToPILImage()(img.to('cpu'))
+        axs[0, i].imshow(np.asarray(img))
+        axs[0, i].set(xticklabels=[], yticklabels=[], xticks=[], yticks=[])
+
+
+####################################
+# The :func:`~torchvision.io.read_image` function allows to read an image and
+# directly load it as a tensor
+
+dog1 = read_image(str(Path('assets') / 'dog1.jpg'))
+dog2 = read_image(str(Path('assets') / 'dog2.jpg'))
+show([dog1, dog2])
+
+####################################
+# Transforming images on GPU
+# --------------------------
+# Most transforms natively support tensors on top of PIL images (to visualize
+# the effect of the transforms, you may refer to see
+# :ref:`sphx_glr_auto_examples_plot_transforms.py`).
+# Using tensor images, we can run the transforms on GPUs if cuda is available!
+
+import torch.nn as nn
+
+transforms = torch.nn.Sequential(
+    T.RandomCrop(224),
+    T.RandomHorizontalFlip(p=0.3),
+)
+
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
+dog1 = dog1.to(device)
+dog2 = dog2.to(device)
+
+transformed_dog1 = transforms(dog1)
+transformed_dog2 = transforms(dog2)
+show([transformed_dog1, transformed_dog2])
+
+####################################
+# Scriptable transforms for easier deployment via torchscript
+# -----------------------------------------------------------
+# We now show how to combine image transformations and a model forward pass,
+# while using ``torch.jit.script`` to obtain a single scripted module.
+#
+# Let's define a ``Predictor`` module that transforms the input tensor and then
+# applies an ImageNet model on it.
+
+from torchvision.models import resnet18
+
+
+class Predictor(nn.Module):
+
+    def __init__(self):
+        super().__init__()
+        self.resnet18 = resnet18(pretrained=True, progress=False).eval()
```

Comment:
How often does this download happen on CI, at every PR or only when we touch `gallery` files?

Reply:
That's a great question and it's something that it still unclear to me.
In theory, this should only need to be downloaded when we modify this example because cross-PR caching is possible with CircleCI, and sphinx-gallery dumps some md5 hashes to avoid recomputing big examples.

In practice I haven't verified yet that this is properly working as it hasn't been a major issue so far. But it will be when we start introducing bigger examples like the tutorials. I'll make sure to check at that point, if that's OK? For now the build_docs CI job is still much faster than the rest so it's not the bottleneck